Sources_v3/StreamChat/Database/DTOs/MessageDTO.swift

```diff
@@ -211,12 +213,18 @@ extension NSManagedObjectContext: MessageDatabaseSession {
         dto.replyCount = Int32(payload.replyCount)
         dto.extraData = try JSONEncoder.default.encode(payload.extraData)
         dto.isSilent = payload.isSilent
-        dto.reactionScores = payload.reactionScores
         dto.channel = ChannelDTO.loadOrCreate(cid: cid, context: self)
         
         let user = try saveUser(payload: payload.user)
         dto.user = user
         
+        try (payload.latestReactions + payload.ownReactions).forEach {
+            try saveReaction(payload: $0)
+        }
+        dto.reactionScores = payload.reactionScores.reduce(into: [:]) { result, reaction in
+            result[reaction.key.rawValue] = reaction.value
+        }
```

Comment:
This is here just to convert `String` -> `ReactionType`? If so, it'd be great to add a comment explaining that.

Reply:
@b-onc already mentioned it'd nice to have a helper. I'm thinking about the extension for `Dictionary` type with `mapKeys` method.

Reply:
`mapKeys` sounds great 👍

Reply:
see https://github.com/GetStream/stream-chat-swift/pull/565#discussion_r512732433